mod/src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/ScoreCalculation.kt

```diff
@@ -80,11 +82,59 @@ object ScoreCalculation {
         "default" to FloorRequirement()
     )
 
+    // Raw vars parsed from the dungeon
     // clear stuff
-    var completedRooms = BasicState(0)
-    var clearedPercentage = BasicState(0)
-    val totalRoomMap = mutableMapOf<Int, Int>()
-    val totalRooms = (clearedPercentage.zip(completedRooms)).map { (clear, complete) ->
+    private var completedRooms = BasicState(0)
+    private var clearedPercentage = BasicState(0)
+
+    // death stuff
+    private var deaths = BasicState(0)
+    var firstDeathHadSpirit = BasicState(false)
+
+    // puzzle stuff
+    private var puzzleCount = 0
+    private var completedPuzzleCount = 0
+    private var missingPuzzles = BasicState(0).also {
+        it.onSetValue {
+            printDevMessage("missing puzzles $it", "scorecalcpuzzle")
+        }
+    }
+    private val failedPuzzlesNames = hashSetOf<String>()
+    private var failedPuzzles: BasicState<Int> = BasicState(0).also {
+        it.onSetValue {
+            updateText(totalScore.get())
+        }
+    }
+
+    // secrets stuff
+    var floorReq = BasicState(floorRequirements["default"]!!)
+    private var foundSecrets: BasicState<Int> = BasicState(0).also {
+        it.onSetValue {
+            updateText(totalScore.get())
+        }
+    }
+    private var totalSecrets = BasicState(0)
+
+    // speed stuff
+    private var secondsElapsed = BasicState(0.0)
+
+    // bonus stuff
+    private var crypts = BasicState(0)
+    var mimicKilled = BasicState(false)
+    private var isPaul = BasicState(false)
+
+    init {
+        tickTimer(5, repeats = true) {
+            isPaul.set(
+                (MayorInfo.currentMayor == "Paul" && MayorInfo.mayorPerks.contains("EZPZ")) || MayorInfo.jerryMayor?.name == "Paul"
+            )
+        }
+    }
```

Comment:
This shouldn't change that often so it should be on at least a 20 tick delay in checking. Probably more around 30 seconds ish would make sense.

Reply:
This was just moved and not changed with the pr, it could be delayed to 60 seconds since that's the delay of the MayorInfo timer